src/openseadragon.js

```diff
@@ -2332,6 +2354,62 @@ function OpenSeadragon( options ){
             return $.createAjaxRequest( local );
         },
 
+        /**
+         * An object holding the AJAX queue state.
+         * @static
+         * @type {Object}
+         * @property {Array} requestFuncs Array of functions representing queued AJAX requests
+         * @property {Array} numIncompleteRequests The number of incomplete AJAX requests, including queued and active (but not completed) requests
+         * @property {Array} numActiveRequests The number of AJAX requests currently in-flight
+         */
+        AjaxQueue: {
+            requestFuncs: [],
+            numIncompleteRequests: 0,
+            numActiveRequests: 0,
+        },
+
+        /**
+         * Queues an AJAX request.
+         * @param {XMLHttpRequest} request The request to be queued
+         * @param {Function} sendRequestFunc A function which, when called, will send the request
+         */
+        queueAjaxRequest: function (request, sendRequestFunc) {
+            if(!$.settings.ajaxLoaderLimit) {
+                sendRequestFunc();
+            }
+
+            var oldOnStateChange = request.onreadystatechange;
+
+            var onCompleteRequest = function() {
+                $.AjaxQueue.numIncompleteRequests--;
+
+                if (
+                    $.AjaxQueue.numActiveRequests === $.settings.ajaxLoaderLimit &&
+                    $.AjaxQueue.requestFuncs.length > 0
+                ) {
+                    $.AjaxQueue.requestFuncs.shift()();
+                } else {
+                    $.AjaxQueue.numActiveRequests--;
+                }
+            };
+
+            request.onreadystatechange = function() {
+                if ( request.readyState === 4 ) {
+                    onCompleteRequest();
+                    oldOnStateChange();
+                }
+            };
+
+            $.AjaxQueue.numIncompleteRequests++;
```

Comment:
This value is manipulated but never used. Maybe we don't need it? I suppose it could be nice to have for debugging...